modin/pandas/groupby.py

```diff
@@ -930,138 +1020,103 @@ def _wrap_aggregation(
         ----------
         qc_method : callable
             The query compiler method to call.
-        default_func : callable
-            The function to call if we need to default to pandas.
-        drop : bool, default: True
-            Whether or not to the grouping columns should be dropped on this operation.
-        numeric_only : bool, default: True
+        numeric_only : bool, default: None
             True for numeric only computations, False otherwise.
+        agg_args : list-like, optional
+            Positional arguments to pass to the aggregation function.
+        agg_kwargs : dict-like, optional
+            Keyword arguments to pass to the aggregation function.
         **kwargs : dict
-            The keyword arguments to be passed to the calling function.
+            Keyword arguments to pass to the specified query compiler's method.
 
         Returns
         -------
         DataFrame or Series
             Returns the same type as `self._df`.
         """
-        if self._axis != 0:
-            return self._default_to_pandas(default_func, **kwargs)
-        # For aggregations, pandas behavior does this for the result.
-        # For other operations it does not, so we wait until there is an aggregation to
-        # actually perform this operation.
-        if not self._is_multi_by and drop and self._drop and self._as_index:
-            groupby_qc = self._query_compiler.drop(columns=self._by.columns)
+        agg_args = tuple() if agg_args is None else agg_args
+        agg_kwargs = dict() if agg_kwargs is None else agg_kwargs
+
+        if numeric_only is None:
+            # pandas behaviour: if `numeric_only` wasn't explicitly specified then
+            # the parameter is considered to be `False` if there are no numeric types
+            # in the frame and `True` otherwise.
+            numeric_only = any(
+                is_numeric_dtype(dtype) for dtype in self._query_compiler.dtypes
+            )
```

Comment:
the computation of the `.dtypes` property is deferred most of the time, the triggering of its computation would cost us the same [as a regular MapReduce function](https://github.com/modin-project/modin/blob/67013f935861187c784415d4265a5b437dba5690/modin/core/dataframe/pandas/dataframe/dataframe.py#L167-L189) which is kinda sad, maybe we can try to ignore the `numeric_only` parameter at all?

Reply:
But we can't ignore this parameter, because it's need for correct computation by partition.groupby internally (see https://github.com/modin-project/modin/pull/3579#discussion_r755105520).

Reply:
Indeed, let's file a separate tracker (and stop worrying about it in this PR, it's already huge).